Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,32 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x) {
+    if (!x) {
+        return 0;
+    }
+#if (defined(__clang__) || defined(__GNUC__))
+    return sizeof(unsigned long) * 8 - __builtin_clzl(x);
+#elif defined(_MSC_VER)
+    Py_BUILD_ASSERT(4 == sizeof(unsigned long));
+    unsigned long msb;
+    _BitScanReverse(&msb, x);
+    return 32 - msb;
+#else
+    int msb = 0;
+    while (x != 0) {
+        msb += 1;
+        x >>= 1;
+    }
+    return msb;
```

Comment:
Can't we keep BitLengthTable? Maybe define the table directly in the function.

Reply:
Since `_Py_bit_length` is now `inline`, this would mean that the table is duplicated to every compilation unit that includes it. That's not many, currently, so maybe that's acceptable. Then again, this fallback will probably not be used anyway. Thoughts?

Reply:
If you don't want it to be duplicated, use `PyAPI_DATA()` to declare the table and rename it to `_Py_bit_length_table`. IMO it's better to always declare the table, since C extensions can be built by a different compiler than Python itself.

Reply:
Nvm, don't think the duplication is a real issue.